src/Symfony/Installer/Application.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/*
+ * This file is part of the Symfony Installer package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Installer;
+
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Application as ConsoleApplication;
+
+/**
+ * This is main Symfony Installer console application class.
+ */
+class Application extends ConsoleApplication
+{
+    const VERSIONS_URL = 'https://get.symfony.com/symfony.version';
+
+    public function doRun(InputInterface $input, OutputInterface $output)
+    {
+        $commandName = $this->getCommandName($input);
+
+        if ($this->isInPharMode() && !in_array($commandName, array('self-update', 'selfupdate'), true)) {
```

Comment:
Just asking: shouldn't we also avoid to call this command for `list`, `help` and the other common actions? Maybe we could replace the `!in_array($commandName, array(...))` by `in_array($commandName, array('new', 'demo'))` ?